src/Cache/CacheOperationTypes/CacheOperationTypes.php

```diff
@@ -322,3 +323,121 @@ class CacheDeleteResponseError extends CacheDeleteResponse
 {
     use ErrorBody;
 }
+
+abstract class CacheListFetchResponse extends ResponseBase
+{
+    public function asHit() : CacheListFetchResponseHit|null
+    {
+        if ($this->isHit())
+        {
+            return $this;
+        }
+        return null;
+    }
+
+    public function asMiss() : CacheListFetchResponseMiss|null
+    {
+        if ($this->isMiss())
+        {
+            return $this;
+        }
+        return null;
+    }
+
+    public function asError() : CacheListFetchResponseError|null
+    {
+        if ($this->isError())
+        {
+            return $this;
+        }
+        return null;
+    }
+}
+
+class CacheListFetchResponseHit extends CacheListFetchResponse
+{
+
+    private array $values = [];
+
+    public function __construct(_ListFetchResponse $response)
+    {
+        parent::__construct();
+        if ($response->getFound())
+        {
+            foreach ($response->getFound()->getValues() as $value)
+            {
+                $this->values[] = $value;
```

Comment:
is this how you append to a list in PHP?  🤯 seems weird to be doing this in a loop

Reply:
Yes, this is the syntax to append to a list. `getValues()` returns protobuf repeated fields that (as far as I know) I have to iterate over like this to extract the actual string values out of. This is not equivalent to `$this->values = $response->getFound()->getValues();` which returns a `TypeError`.

Reply:
yeah the iteration part makes sense to me if this is how you append to a list in php.  that is just, kind of the craziest syntax i've ever seen for appending to a list.  🤯